### PR TITLE
Example app for GetSmoObject

### PR DESCRIPTION
**Pull request by shueybubbles**
Adding an example of some useful core SMO functionality - scripting an object from a URN and capturing the TSQL queries used to generate the script.

### CODE REVIEW

File: examples/GetSmoObject/Program.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            var logger = new SqlClientEventLogger();
+            try
+            {
+                using (var connection = new SqlConnection(new SqlConnectionStringBuilder()
+                {
+                    DataSource = args[0],
+                    InitialCatalog = args[1],
+                    Authentication = SqlAuthenticationMethod.ActiveDirectoryInteractive
+                }.ConnectionString))
```
Comment (Matteo-T) [inline examples/GetSmoObject/Program.cs:31]: Something in the indentation is confusing... or maybe I'm just too sleepy. :)

File: examples/GetSmoObject/SqlClientEventLogger.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            if (eventData.Payload != null)
+            {
+                var now = DateTime.UtcNow.ToString("o");
+                logs = logs.Append(string.Join(Environment.NewLine, eventData.Payload.Select(p => $"{now} [{Environment.CurrentManagedThreadId}]: {p}")) + Environment.NewLine);
+            }
+        }
+        public override string ToString()
+        {
+            return logs.ToString();
+        }
```
Comment (Matteo-T) [inline examples/GetSmoObject/SqlClientEventLogger.cs:37]: public override string ToString() => logs.ToString();

File: examples/GetSmoObject/SqlClientEventLogger.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+
+namespace GetSmoObject
+
+{
```
Comment (Matteo-T) [inline examples/GetSmoObject/SqlClientEventLogger.cs:11]: Remove blank

File: examples/GetSmoObject/SqlClientEventLogger.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (eventData.Payload != null)
+            {
+                var now = DateTime.UtcNow.ToString("o");
+                logs = logs.Append(string.Join(Environment.NewLine, eventData.Payload.Select(p => $"{now} [{Environment.CurrentManagedThreadId}]: {p}")) + Environment.NewLine);
+            }
+        }
+        public override string ToString()
```
Comment (Matteo-T) [inline examples/GetSmoObject/SqlClientEventLogger.cs:34]: Add a blank

**Comment by Matteo-T** [comment]
Do we need a README.md that explains why/when I should be using GetSmoObject?

**Review by Matteo-T — CHANGES_REQUESTED** [review]
See my comments.

**Comment by shueybubbles** [comment]
I hadn't hit Save on the readme before I committed

File: examples/GetSmoObject/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Overview
```
Comment (Matteo-T) [inline examples/GetSmoObject/README.md:1]: Nice!

**Review by Matteo-T — APPROVED** [review]
:shipit:

**Pull request merged**